### PR TITLE
KAFKA-9405: Use Map API computeIfAbsent Where Applicable

### PR DESCRIPTION
**Pull request by belugabehr**
*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java [hunk 1]
```diff
@@ -2242,11 +2242,10 @@ public DescribeReplicaLogDirsResult describeReplicaLogDirs(Collection<TopicParti
 
         Map<Integer, Set<TopicPartition>> partitionsByBroker = new HashMap<>();
 
-        for (TopicPartitionReplica replica: replicas) {
-            if (!partitionsByBroker.containsKey(replica.brokerId()))
-                partitionsByBroker.put(replica.brokerId(), new HashSet<>());
-            partitionsByBroker.get(replica.brokerId()).add(new TopicPartition(replica.topic(), replica.partition()));
-        }
+		for (TopicPartitionReplica replica : replicas) {
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java]: Can you please fix the indenting here and below?

Comment (belugabehr) [reply clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java]: @ijuma Thanks for the review! Yes.  Of course.  Woops.  Sorry about that.

**Comment by belugabehr** [comment]
@ijuma Can you trigger a test here?  Thanks!

**Comment by ijuma** [comment]
ok to test

**Review by ijuma — APPROVED** [review]
LGTM, thanks.

**Comment by ijuma** [comment]
One job passed and one had 1 flaky test failure.

**Pull request merged**